services/cd-service/pkg/notify/notify.go

```diff
@@ -31,25 +34,55 @@ func (n *Notify) Subscribe() (<-chan struct{}, Unsubscribe) {
 
 	n.mx.Lock()
 	defer n.mx.Unlock()
-	if n.listener == nil {
-		n.listener = map[chan struct{}]struct{}{}
+	if n.oveviewListener == nil {
+		n.oveviewListener = map[chan struct{}]struct{}{}
 	}
 
-	n.listener[ch] = struct{}{}
+	n.oveviewListener[ch] = struct{}{}
 	return ch, func() {
 		n.mx.Lock()
 		defer n.mx.Unlock()
-		delete(n.listener, ch)
+		delete(n.oveviewListener, ch)
 	}
 }
 
 func (n *Notify) Notify() {
 	n.mx.Lock()
 	defer n.mx.Unlock()
-	for ch := range n.listener {
+	for ch := range n.oveviewListener {
 		select {
 		case ch <- struct{}{}:
 		default:
 		}
 	}
 }
+
+type ChangedAppNames []string
+
+func (n *Notify) SubscribeChangesApps() (<-chan ChangedAppNames, Unsubscribe) {
+	ch := make(chan ChangedAppNames, 1)
+	ch <- ChangedAppNames{}
+
+	n.mx.Lock()
+	defer n.mx.Unlock()
+	if n.changeAppsListener == nil {
+		n.changeAppsListener = map[chan ChangedAppNames]ChangedAppNames{}
+	}
+	n.changeAppsListener[ch] = ChangedAppNames{}
+	return ch, func() {
+		n.mx.Lock()
+		defer n.mx.Unlock()
+		delete(n.changeAppsListener, ch)
+	}
+}
+
+func (n *Notify) NotifyChangedApps(changedApps ChangedAppNames) {
+	n.mx.Lock()
+	defer n.mx.Unlock()
+	for ch := range n.changeAppsListener {
+		select {
+		case ch <- changedApps:
```

Comment:
This doesn't work here. The original overview notifier was able to simply skip updates for slow subscribers because it always sends out the entire overview. If you want to send partial updates, you have to make sure that partial updates don't get lost.

Reply:
yes, what's missing is sending all apps, whenever a client connects to the streaming grpc call.